src/expire.c

```diff
@@ -258,8 +258,8 @@ void activeExpireCycle(int type) {
                     if (table == 1 && !dictIsRehashing(db->expires)) break;
 
                     unsigned long idx = db->expires_cursor;
-                    idx &= db->expires->ht[table].sizemask;
-                    dictEntry *de = db->expires->ht[table].table[idx];
+                    idx &= DICTHT_SIZE_MASK(db->expires->ht_size_exp[table]);
+                    dictEntry *de = db->expires->ht_table[table][idx];
```

Comment:
While we're touching this, can we hide this with a dict function?

Reply:
?

Reply:
You mean something like:
```c
#define dictGetEntry(d,t,h) (d)->ht_table[t][(h) & DICTHT_SIZE_MASK((d)->ht_size_exp[t])];
```
And use it in all relevant places?

Reply:
@madolson ?